Sources/PackageGraph/PubGrub/ContainerProvider.swift

```diff
@@ -79,7 +82,7 @@ final class ContainerProvider {
             // Otherwise, fetch the container from the provider
             self.underlying.getContainer(
                 for: package,
-                skipUpdate: self.skipUpdate,
+                updateStrategy: self.skipUpdate ? .never : .always, // FIXME
```

Comment:
it could be nice optimization if we could use "ifNeeded" in PubGrub too, but I am not sure it would be correct. @neonichu ideas?

Reply:
I think we probably need something more elaborate there. If there's an existing pin, I believe we try to stick to it in PubGrub, but we would need the ability to switch the strategy if needed.

Reply:
left this for follow up PR